radar/languages-and-frameworks/typescript.md

```diff
@@ -0,0 +1,27 @@
+---
+title: "Typescript"
+ring: adopt
+quadrant: "languages-and-frameworks"
+tags: [typescript]
+---
+
+[TypeScript](https://www.typescriptlang.org/) is a superset of JavaScript that adds static typing to the language. It transpiles to JavaScript, meaning that an application written in Typescript will be executed in the a Javascript runtime. The JavaScript runtime, whether it's a browser environment or Node.js, doesn't have direct awareness of the original TypeScript source code. It treats the transpiled JavaScript code just like any other JavaScript code.
+
+The purpose of using Typescript is for making it easier for developers to write safer and more maintainable code.
+This is achieved by influencing the developer experience with several factors:
+
+1. **Types help describe code**: Types act as a form of documentation within the code, providing explicit information about the expected data structures and formats. Developers can quickly understand the purpose and usage of functions and variables just by looking at their type annotations. This does not imply comments become obsolete; on the opposite, they can focus on explaing the design choices.
```

Comment:
type annotations also help increase DX because they allow IDEs to have better code autocompletion

Reply:
So we can add a page for type annotations 😉

Reply:
imho type annotation are an indistinguishable part of typescript as a language, I wouldn't put them as a separate entry

Reply:
Sorry my bad, I thought you were talking about "using type annotations on JavaScript" for better IDE DX. 

So I don't understand the comment

Reply:
No sorry, I wasn't clear

I was saying: an added benefit of TypeScript is that thanks to type annotations not only can you immediately notice errors, but you also have better code autocompletion

Reply:
ahhh ok, I see. Good, I'll add the benefit